### PR TITLE
Move manylinux1 precompiler definitions into port_platform.h

### PR DESCRIPTION
**Pull request by soltanmm**
Moves the `manylinux1` precompiler definitions into `port_platform.h`.

I'll unmark `DO NOT MERGE` after seeing the distrib test results on Jenkins.

cc @nicolasnoble (I'ma assign you after seeing results, so please take a quick look to short-circuit anything I'm doing)
cc @jtattermusch 

<sup>EDIT: now following [this build](https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_artifacts/730/)</sup>

### CODE REVIEW

**Comment by jtattermusch** [comment]
Looks like the same amount of distribtests is succeeding as before.

**Comment by jtattermusch** [comment]
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_distribtest/663/architecture=x64,language=python,platform=linux/console

**Comment by jtattermusch** [comment]
@nicolasnoble  let's merge this?

File: include/grpc/impl/codegen/port_platform.h [hunk 1]
```diff
@@ -114,6 +114,38 @@
 #define GPR_WIN32_ATOMIC 1
 #define GPR_MSVC_TLS 1
 #endif
+#elif defined(GPR_MANYLINUX1)
+// TODO(atash): manylinux1 is just another __linux__ but with ancient
+// libraries; it should be integrated with the `__linux__` definitions below.
```
Comment (nicolasnoble) [inline include/grpc/impl/codegen/port_platform.h]: Re: the comment: do you have a "proper" way to detect "manylinux" from within __linux__ ? And in fact, a good portion of these could be folded inside __linux__ using GPR_MANYLINUX1 as you just said - it just needs to be detected, either manually using GPR_MANYLINUX1, or any other definition that your compiler would do. Either way, this is fine.

Comment (soltanmm) [inline include/grpc/impl/codegen/port_platform.h]: `manylinux1` is described as:
```
GLIBC <= 2.5
CXXABI <= 3.4.8
GLIBCXX <= 3.4.9
GCC <= 4.2.0
```
What's missing for me to Nike this is a map between all the feature `#define`s used by gRPC core and the versions of `glibc` and `libgcc`.

**Pull request merged**